### PR TITLE
Upgrade OpenCV version

### LINKED ISSUE #556: Upgrade OpenCV Version
It looks like it's time to upgrade our OpenCV dependency from `opencv-python==4.1.2.30`, for the following reasons:
- Allow PeekingDuck to run on M1 Mac
- Address behavioural differences with newer opencv versions, such as `cv2.resizeWindow()` and input arg types for coordinates when using draw functions

We should upgrade to version 4.5 and newer, as there are some issues with the prior versions as documented below. As of now, the latest `opencv-python==4.5.4.60` still works with Python 3.6, which is the lowest version of Python supported by PeekingDuck.

| Version | Comments |
| - | - |
| 4.1.2.30 | Recommended openCV version for PeekingDuck as of 15 Dec 2021. Doesn't work for M1 Mac. |
| From 4.2.0.32 to 4.4.0.46 | Not recommended to use these versions due to a few reported issues mainly due to QT issues with MacOS, such as [`imshow()` not showing](https://github.com/opencv/opencv-python/issues/423), or [segmentation fault](https://github.com/opencv/opencv/issues/18079), or [QT error in MacOS](https://github.com/opencv/opencv-python/issues/290)
| 4.5.1.48 | Works with M1 Mac and the issues from earlier versions have been fixed by dropping QT support for Mac. Different behaviour for `cv2.resize()` and `cv2.resize_window()` when compared to `4.1.2.30`. |
| 4.5.2.52 | Coordinates need to be `int` instead of `float` for drawing |
| 4.5.2.54 | Bugfix for [4.5.2.52](https://github.com/opencv/opencv-python/releases?page=1) |

Definition of done:
- [ ] Update `requirements.txt` and `setup.cfg` with the new version
- [ ] Update `draw` nodes to take in `int` coordinates.

### PR DESCRIPTION
**Pull request by leeping-ng**
Closes #556. 

- `4.5.2.54` chosen due to factors mentioned in the issue.
- unpinned dependency version as a [good practice](https://stackoverflow.com/questions/28509481/should-i-pin-my-python-dependencies-versions) for releasing packages

### CODE REVIEW

File: peekingduck/pipeline/nodes/draw/utils/bbox.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def _draw_tag(
     )
     bbox_width = btm_right[0] - top_left[0]
     offset = int((bbox_width - text_width) / 2)
-    position = (int(top_left[0] + offset), int(top_left[1] - baseline))
+    position = (top_left[0] + offset, int(top_left[1] - baseline))
```
Comment (ongtw) [inline peekingduck/pipeline/nodes/draw/utils/bbox.py:165]: `cv2.getTextSize()` returns `(text_width, text_height), baseline`, all of which are integers.
So there is no need to `int()` once the coords are integers.
The same applies to line 111 in `bbox.py` for `_draw_label()`.

Verified from opencv [C++ docs](https://shimat.github.io/opencvsharp_docs/html/66fb2360-14d2-3431-c0ef-1679c153cf06.htm), [source](https://github.com/opencv/opencv/blob/4223495e6cd67011f86b8ecd9be1fa105018f3b1/modules/imgproc/include/opencv2/imgproc.hpp#L4770), and runtime printouts:
`*getTextSize: text_width=<class 'int'>, text_height=<class 'int'>, baseline=<class 'int'>`

File: peekingduck/pipeline/nodes/draw/utils/general.py [hunk 2]
```diff
@@ -59,4 +59,4 @@ def project_points_onto_original_image(
     projected_points[:, 0] *= width
     projected_points[:, 1] *= height
 
-    return projected_points
+    return projected_points.astype(int)
```
Comment (ongtw) [inline peekingduck/pipeline/nodes/draw/utils/general.py:62]: Just a note that `numpy.astype()` does not round up the numbers:
```
>>> import numpy as np
>>> x = np.array([1.1, 2.5, 3.7])
>>> x
array([1.1, 2.5, 3.7])
>>> x.astype(int)
array([1, 2, 3])
```

If intention is to round, can do this:
```
>>> np.round(x).astype(int)
array([1, 2, 4])
```

**Review by ongtw — COMMENTED** [review]
Two minor technical notes for consideration.

File: peekingduck/pipeline/nodes/draw/utils/bbox.py (hunk 1 above)
Comment (leeping-ng) [reply peekingduck/pipeline/nodes/draw/utils/bbox.py]: Wasn't able to find this, thanks! Updated.

File: peekingduck/pipeline/nodes/draw/utils/general.py (hunk 2 above)
Comment (leeping-ng) [reply peekingduck/pipeline/nodes/draw/utils/general.py]: Implemented rounding for that extra bit of accuracy.

**Review by ongtw — APPROVED** [review]
Good to go.

**Pull request merged**